packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -619,6 +768,82 @@ enum SelectionStatus {
   none,
 }
 
+/// The details of a selection.
+///
+/// This object is created by users of the [SelectionListenerSelectionChangedCallback] callback.
+///
+/// This includes information such as the status of the selection
+/// indicating if it is collapsed or uncollapsed, the start and end offsets
+/// of the selection local to the [SelectionListener] that reports this object,
+/// and whether the selection is ongoing.
+///
+/// See also:
+///
+///   * [SelectionListener], which provides a [SelectionDetails] object
+///   for the selection under its subtree in its [SelectionListener.onSelectionChanged]
+///   callback.
+@immutable
+class SelectionDetails {
+  /// Creates a selection details object.
+  const SelectionDetails({
+    required this.status,
+    required this.selectionFinalized,
+    required this.globalStartOffset,
+    required this.globalEndOffset,
+  });
+
+  /// The status of ongoing selection under the [Selectable]
+  /// or [SelectionHandler] that created this object.
+  final SelectionStatus status;
+
+  /// Whether the selection is ongoing.
+  ///
+  /// Returns false if the selection is ongoing and
+  /// true if the selection is finalized.
+  final bool selectionFinalized;
+
+  /// The global start offset.
```

Comment:
I don't think this is "global" within the app right? This should probably use a more accurate term if that's the case.

Reply:
Also consider using existing classes `TextRange`.

Reply:
Changed to `local*Offset`, but open to other suggestions.